c10/xpu/XPUFunctions.cpp

```diff
@@ -41,15 +42,19 @@ void enumDevices(std::vector<std::unique_ptr<sycl::device>>& devices) {
     auto device_list = platform.get_devices();
     for (const auto& device : device_list) {
       if (device.is_gpu()) {
-        devices.push_back(std::make_unique<sycl::device>(device));
+        gDevicePool.devices.push_back(std::make_unique<sycl::device>(device));
       }
     }
   }
+  // We need to check the number of XPU devices to prevent DeviceIndex overflow.
+  TORCH_CHECK(
```

Comment:
This function `enumDevices` could be called only once when XPU runtime is initialized. So we put the check for DeviceIndex overflow here. Meanwhile, in our design, XPU runtime doesn't depend on `c10::Device::MAX_NUM_DEVICES`. So here only check the number of XPU devices should not be greater than std::numeric_limits<DeviceIndex>::max(),